### PR TITLE
[Dynamo][4/N] Enable clang-tidy coverage on torch/csrc/dynamo/*

### PR DESCRIPTION
**Pull request by cyyever**
This PR enables clang-tidy coverage on torch/csrc/dynamo/* and also contains other small improvements.
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: .lintrunner.toml [hunk 1]
```diff
@@ -224,8 +224,8 @@ exclude_patterns = [
     'third_party/**/*',
     'torch/csrc/api/**',
     'torch/csrc/autograd/generated/**',
-    'torch/csrc/dynamo/*',
     'torch/csrc/distributed/**/*',
+    'torch/csrc/dynamo/eval_frame.h',
```
Comment (Skylion007) [inline .lintrunner.toml:228]: Why ignore this one header?

Comment (cyyever) [reply .lintrunner.toml:228]: Because clang-tidy reports a strange warning. Let's re-check it.

**Pull request closed without merging**